### PR TITLE
[Merged by Bors] - fix: sync file list expected by cache with Lake changes

### PR DESCRIPTION
**Pull request by Kha**
The `.c.trace` files no longer exist, see https://github.com/leanprover/lake/commit/6499494befc24e388d12f3d9411f99aae0769491#diff-43ead6410f2dcb31276401eb765b8ada18c8e288626dfa8d6123cfab46961e7aL81-L83.

---

Let's see if it uploads something now

### CODE REVIEW

**Comment by Kha** [comment]
@arthurpaulino Is there any conceivable scenario where `allExist` failing should *not* result in a fatal error?

**Comment by eric-wieser** [comment]
I assume the idea is to allow uploading a cache for an incomplete build that has successfully built some files? Either way, that behavior sounds like it should be hidden behind a flag.

**Comment by Ruben-VandeVelde** [comment]
> I assume the idea is to allow uploading a cache for an incomplete build that has successfully built some files? Either way, that behavior sounds like it should be hidden behind a flag.

This is really important if you need to edit a low-level file while porting a new file, for example.

**Comment by eric-wieser** [comment]
Ah good point; in CI you want a partial cache to be uploaded when a PR build fails, so that you don't have to start from scratch.

**Review by eric-wieser — APPROVED** [review]
Let's wait for the new cache to be uploaded before merging

**Comment by eric-wieser** [comment]
bors d+

Feel free to merge if you don't want to wait for further feedback. No need to wait for the remaining CI, since the cache upload is already done.

**Comment by Kha** [comment]
Since there is no hurry, I'll leave taking a final look to @gebner

**Comment by Kha** [comment]
> in CI you want a partial cache to be uploaded when a PR build fails, so that you don't have to start from scratch

For that purpose, I believe it would be better to say "upload a module's files iff the .olean file exists". This logic would be compatible with producing optional output files such as with #3575 or #3404 (/cc @semorrison).

**Comment by Kha** [comment]
Well, I guess we're preventing new files from being uploaded, so let's get this merged. My first bors command, yay.

bors r+

**Pull request closed without merging**
